extensions/WSLExtension/DistributionDefinitions/DistributionDefinitionHelper.cs

```diff
@@ -0,0 +1,112 @@
+/// <see cref="Constants.KnownDistributionsWebJsonLocation"/>.
+/// </summary>
+public class DistributionDefinitionHelper : IDistributionDefinitionHelper
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(DistributionDefinitionHelper));
+
+    private readonly IHttpClientFactory _httpClientFactory;
+
+    private readonly PackageHelper _packageHelper = new();
+
+    private readonly JsonSerializerOptions _jsonOptions = new()
+    {
+        PropertyNameCaseInsensitive = true,
+        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+    };
+
+    public DistributionDefinitionHelper(IHttpClientFactory httpClientFactory)
+    {
+        _httpClientFactory = httpClientFactory;
+    }
+
+    /// <inheritdoc cref="IDistributionDefinitionHelper.GetDistributionDefinitionsAsync"/>
+    public async Task<Dictionary<string, DistributionDefinition>> GetDistributionDefinitionsAsync()
+    {
+        try
+        {
+            // Get the update to date distribution definitions from WSL GitHub repository.
+            var client = _httpClientFactory.CreateClient();
+            var distributionDefinitionsJson = await client.GetStringAsync(KnownDistributionsWebJsonLocation);
+            var webDefinitions = JsonSerializer.Deserialize<DistributionDefinitions>(distributionDefinitionsJson, _jsonOptions);
+            var distributionDefinitionsMap = new Dictionary<string, DistributionDefinition>();
+
+            foreach (var definition in webDefinitions!.Values)
+            {
+                // Only supported distributions for this machine.
+                if (ShouldAddDistribution(definition))
+                {
+                    distributionDefinitionsMap.Add(definition.Name, definition);
+                }
+            }
+
+            // Merge the local distribution information we have stored in DistributionDefinition.yaml with the one above.
+            var uri = new Uri(KnownDistributionsLocalYamlLocation);
+            var storageFile = await StorageFile.GetFileFromApplicationUriAsync(uri);
+            var localYamlDefinitionsFile = await FileIO.ReadTextAsync(storageFile);
+            var localYamlDefinitions = BuildYamlDeserializer().Deserialize<List<DistributionDefinition>>(localYamlDefinitionsFile);
+            foreach (var localYamlDefinition in localYamlDefinitions)
+            {
+                // Ignore distributions that we have in the local yaml file but are no longer present in the web file.
+                if (!distributionDefinitionsMap.TryGetValue(localYamlDefinition.Name, out var definitionFromWeb))
+                {
+                    continue;
+                }
+
+                definitionFromWeb.Publisher = localYamlDefinition.Publisher;
+                definitionFromWeb.WindowsTerminalProfileGuid = localYamlDefinition.WindowsTerminalProfileGuid;
+
+                // Ignore distributions in the local file like oracle Linux without an image.

```

Comment:
the web definitions are here: https://aka.ms/wsldistributionsjson, we use these first. The list we iterate through next is the local yaml file to augment the info we get from the web. For now, we know if it has no image then there is no reason to do any work with it. (only ones without an image are the oracle ones). In the future, if we decide to add more metadata to the definitions in the yaml file, then we can look into not ignoring the ones without an image. Currently I just ignore them because I know for a fact that there is no work to be done for them at this point in the loop